src/D2L.CodeStyle.TestAnalyzers/ExpectedException/ExpectedExceptionAnalyzer.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.TestAnalyzers.ExpectedException {
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	public sealed class ExpectedExceptionAnalyzer : DiagnosticAnalyzer {
+		public const string DiagnosticId = "D2L0004";
+		private const string Category = "Safety";
+
+		private const string Title = "Ensure test does not contain ExpectedException.";
+		private const string Description = "'ExpectedException' should not be used in tests.";
+		internal const string MessageFormat = "'ExpectedException' is not safe for use, no longer available in NUnit 3";
+
+		private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+			DiagnosticId,
+			Title,
+			MessageFormat,
+			Category,
+			DiagnosticSeverity.Warning,
+			isEnabledByDefault: true,
+			description: Description
+		);
+		
+		public override void Initialize(AnalysisContext context)
+		{
+			context.RegisterSyntaxNodeAction(
+				AnalyzeSyntaxNode,
+				SyntaxKind.AttributeList
+			);
+		}
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create( Rule );

```

Comment:
nit: I'd move this above `Initialize` for style (and consistency with the other analyzers)